### PR TITLE
Live coding interview using Shared meeting stage - Nodejs

### PR DESCRIPTION
**Pull request by NikitaJain-MSFT**

### CODE REVIEW

File: samples/meetings-live-code-interview/nodejs/ClientApp/src/App.test.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { render, screen } from '@testing-library/react';
```
Comment (AbhijeetThakur-msft) [inline samples/meetings-live-code-interview/nodejs/ClientApp/src/App.test.js:1]: Does this test pass?

File: samples/meetings-live-code-interview/nodejs/ClientApp/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  }
+
+  public getAppDom = () => {
+    const context = getContext({
+      baseFontSize: 10,
+      style: this.state.themeStyle
+    });
+    return (
+      <TeamsThemeContext.Provider value={context}>
+        <Suspense fallback={<div></div>}>
```
Comment (AbhijeetThakur-msft) [inline samples/meetings-live-code-interview/nodejs/ClientApp/src/App.tsx:101]: for fallback, user will not know what happened, if needed provide some message.

File: samples/meetings-live-code-interview/nodejs/ClientApp/src/components/question.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import * as microsoftTeams from "@microsoft/teams-js";
+import { IQuestionDetails } from '../types/question';
+import { Flex, Text } from '@fluentui/react-northstar'
+import Editor from '@monaco-editor/react';
+import io from "socket.io-client";
+import "./tab.css"
+
+const Question = (props: any) => {
+    const params = props.match.params;
+    const questionNumber = params['srno'];
```
Comment (AbhijeetThakur-msft) [inline samples/meetings-live-code-interview/nodejs/ClientApp/src/components/question.tsx:11]: rename this srno to questionId as this is not meanigfull

File: samples/meetings-live-code-interview/nodejs/ClientApp/src/components/question.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    const [data, setData] = React.useState();
+    const socket = io();
+
+    React.useEffect(() => {
+        microsoftTeams.initialize();
+
+    }, [])
+
+    const debounce = (callback: any, lim: any) => {
```
Comment (AbhijeetThakur-msft) [inline samples/meetings-live-code-interview/nodejs/ClientApp/src/components/question.tsx:20]: rename this to give proper understanding and avoid short forms.

File: samples/meetings-live-code-interview/nodejs/ClientApp/src/components/question.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+    // get latest state of editor.
+    socket.on("message", data => {
+        setData(data);
+    });
+
+    return (
+        <>
+            {IQuestionDetails.questions ? IQuestionDetails.questions.map((question) => {
+                if (question.srNo == questionNumber) {
```
Comment (AbhijeetThakur-msft) [inline samples/meetings-live-code-interview/nodejs/ClientApp/src/components/question.tsx:45]: rename srno to questionId or questionNo

File: samples/meetings-live-code-interview/nodejs/ClientApp/src/components/tab.css [hunk 6]
```diff
@@ -0,0 +1,20 @@
+.text-label {
```
Comment (AbhijeetThakur-msft) [inline samples/meetings-live-code-interview/nodejs/ClientApp/src/components/tab.css:1]: check if these all classes are being used.

File: samples/meetings-live-code-interview/nodejs/ClientApp/src/components/tab.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import * as microsoftTeams from "@microsoft/teams-js";
+import { Flex, Button, Text, Card, CardBody, CardHeader } from '@fluentui/react-northstar'
+import "./tab.css"
+import { IQuestionDetails } from '../types/question';
+
+const Tab = () => {
+
```
Comment (AbhijeetThakur-msft) [inline samples/meetings-live-code-interview/nodejs/ClientApp/src/components/tab.tsx:8]: remove this extra space.

File: samples/meetings-live-code-interview/nodejs/ClientApp/src/types/question.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
+export const IQuestionDetails = {
```
Comment (AbhijeetThakur-msft) [inline samples/meetings-live-code-interview/nodejs/ClientApp/src/types/question.ts:1]: Provide some comments on what is this used for.

File: README.md [hunk 9]
```diff
@@ -178,6 +178,7 @@ The [Teams Toolkit](https://marketplace.visualstudio.com/items?itemName=TeamsDev
 |22| Activity feed broadcast |Demonstrating a feature to notify any message to all members of the organisation using activity feed notification .|[View][graph-activity-feed-broadcast#csharp]|[View][graph-activity-feed-broadcast#js]|
 |23|App complete auth|This sample demos authentication feature in bot,tab and messaging extension.|[View][app-complete-auth#cs]|
 |24|App identity linking with sso|This sample demos mapping with aad id, facebook, and google account of user in bot, ME and tab.||[View][app-identity-link-with-sso#nodejs]|
+|24|Live coding interview using Shared meeting stage |This sample demos live coding in a teams meeting.||[View][meetings-live-code-interview#nodejs]|
 
 [app-hello-world#cs]:samples/app-hello-world/csharp
 [app-hello-world#ts]:samples/app-hello-world/nodejs
```
Comment (AbhijeetThakur-msft) [inline README.md:184]: Please make sure we do not have any horizonal scroll bar in any screen.

**Review by AbhijeetThakur-msft — CHANGES_REQUESTED** [review]

File: samples/meetings-live-code-interview/nodejs/api/package.json [hunk 10]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "sequential-workflow-adaptive-card",
+  "version": "1.0.0",
+  "description": "Sequential Workflows in adaptive card",
```
Comment (eoinobrien) [inline samples/meetings-live-code-interview/nodejs/api/package.json:4]: Can we update this to something accurate?

File: samples/meetings-live-code-interview/nodejs/ClientApp/src/types/question.ts [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+            language:"javascript",
+            defaultValue:"const arr = [1,4,5,4,2,1,9,5,7,6,3]; \n\n function getUniqueElement(arr){ \n   \/\/write your code here\n }"
+        },
+        {
+            questionId : 3,
+            question:"Get count of character from string",
+            expectedOuput:"if string is 'javascript' then output will be {j:1,a:2,v:1,s:1,c:1,r:1,i:1,p:1,t:1}",
+            language:"javascript",
+            defaultValue:"const str = 'javascript' \n\n function getCount(str){\n   \/\/ write your code here\n }"
+        }
```
[the remark this answers is not in the document]
Comment (NikitaJain-MSFT) [reply samples/meetings-live-code-interview/nodejs/ClientApp/src/types/question.ts:24]: as per requirements we do not need to add configurable questions.

**Review by v-chetsh — APPROVED** [review]
Changes looks fine for me, approving it.

**Review by Wajeed-msft — APPROVED** [review]

**Review by AbhijeetThakur-msft — APPROVED** [review]

**Pull request merged**